app/scripts/metamask-controller.test.js

```diff
@@ -1287,6 +1287,129 @@ describe('MetaMaskController', () => {
           );
         });
       });
+
+      it.todo(
+        'should only process `metamask-provider` multiplex formatted messages',
+      );
+    });
+
+    describe('#setupUntrustedCommunicationCaip', () => {
+      const mockTxParams = { from: TEST_ADDRESS };
+
+      beforeEach(() => {
+        initializeMockMiddlewareLog();
+        metamaskController.preferencesController.setSecurityAlertsEnabled(
+          false,
+        );
+        jest
+          .spyOn(metamaskController.onboardingController.store, 'getState')
+          .mockReturnValue({ completedOnboarding: true });
+        metamaskController.preferencesController.setUsePhishDetect(true);
+      });
+
+      afterAll(() => {
+        tearDownMockMiddlewareLog();
+      });
+
+      it('adds a tabId, origin and networkClient to requests', async () => {
+        const messageSender = {
+          url: 'http://mycrypto.com',
+          tab: { id: 456 },
+        };
+        const streamTest = createThroughStream((chunk, _, cb) => {
+          if (chunk.data && chunk.data.method) {
+            cb(null, chunk);
+            return;
+          }
+          cb();
+        });
+
+        metamaskController.setupUntrustedCommunicationCaip({
+          connectionStream: streamTest,
+          sender: messageSender,
+        });
+
+        const message = {
+          id: 1999133338649204,
+          jsonrpc: '2.0',
+          params: [{ ...mockTxParams }],
+          method: 'eth_sendTransaction',
+        };
+        await new Promise((resolve) => {
+          streamTest.write(
+            {
+              type: 'caip-x',
+              data: message,
+            },
+            null,
+            () => {
+              setTimeout(() => {
+                expect(loggerMiddlewareMock.requests[0]).toHaveProperty(
+                  'origin',
+                  'http://mycrypto.com',
+                );
+                expect(loggerMiddlewareMock.requests[0]).toHaveProperty(
+                  'tabId',
+                  456,
+                );
+                expect(loggerMiddlewareMock.requests[0]).toHaveProperty(
+                  'networkClientId',
+                  'networkConfigurationId1',
+                );
+                resolve();
+              });
+            },
+          );
+        });
+      });
+
+      it('should add only origin to request if tabId not provided', async () => {
```

Comment:
struggling to see how this test doesn't contradict the one above it?

Reply:
Oh I think I got tripped up on the phrasing here. nevermind

Reply:
Maybe it doesn't matter for this test but I guess `networkClientId` is on this request as well?